lib/spack/spack/util/windows_registry.py

```diff
@@ -59,18 +59,88 @@ def subkeys(self):
     def hkey(self):
         return self._handle
 
+    def OpenKeyEx(self, subname, **kwargs):
+        """Convenience wrapper around winreg.OpenKeyEx"""
+        tty.debug(
+            f"[WINREG ACCESS] Accessing Reg Key {self.path}/{subname} with {kwargs.get('access', 'default')} access"
+        )
+        try:
+            return winreg.OpenKeyEx(self.hkey, subname, **kwargs)
+        except OSError as e:
+            # Expected errors that occur on occasion, these are easily
+            # debug-able and have sufficiently verbose reporting and obvious cause
```

Comment:
IMO it's worth briefly explaining what these error codes mean (which in my case would help me understand why they don't apply for `QueryInfoKey`, for example).

Reply:
I've elaborated on the specific easily diagnoseable errors (and consolidated this similar logic as suggested above).

It's not that these don't apply to `QueryInfoKey`, it's that the errors raised by `QueryInfoKey` haven't been observed to raise the vague, difficult to discern errors the others do, and it interacts with the keys differently (reading metadata vs interacting with they keys directly) but while thinking about it, while I'm centralizing the logic, I might as well include this and future proof.